File: src/handlers/discount_handler.py
import nextcord
from nextcord import Interaction
from nextcord.ext import commands
from src.config.env import DISCOUNT_CHANNEL_ID, SUPABASE_URL, SUPABASE_ANON_KEY
from src.database.supabase import supabase

bot = commands.Bot(command_prefix="!")

@bot.slash_command(name="discount", description="Apply a discount to a customer's order")
async def discount(
    interaction: Interaction,
    order_id: str,
    amount: float,
    customer_id: str,
    reason: str = "No reason provided"
):
    # Fetch original order price from Supabase (optional)
    order = supabase.table("orders").select("*").eq("order_id", order_id).execute()
    if not order.data:
        await interaction.response.send_message(f"❌ Order {order_id} not found.", ephemeral=True)
        return

    original_price = float(order.data[0]["total_price"])
    discounted_price = max(0, original_price - amount)

    # Send plain text message to discount channel
    guild = interaction.guild
    discount_channel = guild.get_channel(DISCOUNT_CHANNEL_ID)
    if discount_channel:
        await discount_channel.send(
            f"💰 Discount Applied!\n"
            f"Order ID: {order_id}\n"
            f"Customer: <@{customer_id}>\n"
            f"Original Price: {original_price}\n"
            f"Discounted Price: {discounted_price}\n"
            f"Reason: {reason}\n"
            f"Discount applied by: {interaction.user.mention}"
        )

    # Notify the customer
    await interaction.response.send_message(f"✅ Discount applied for <@{customer_id}>.", ephemeral=True)

File: src/handlers/discountHandler.js
import { client } from "../bot.js";
import { ENV } from "../config/env.js";
import { supabase } from "../database/supabase.js";

// Handles /discount command for chefs
export const initDiscountHandler = () => {
  client.on("interactionCreate", async (interaction) => {
    if (!interaction.isChatInputCommand()) return;
    if (interaction.commandName !== "discount") return;

    const { options } = interaction;

    try {
      const orderId = options.getString("order_id");
      const amount = options.getNumber("discount_amount");
      const customerId = options.getUser("customer").id;
      const reason = options.getString("reason") || "No reason provided";

      // TODO: Fetch original order price from Supabase
      const originalPrice = 100; // Placeholder for now
      const discountedPrice = originalPrice - amount;

      // Send text message to discount channel
      const discountChannel = await client.channels.fetch(ENV.DISCOUNT_CHANNEL_ID);
      const messageText = `
📢 Discount Applied
Order ID: ${orderId}
Customer: <@${customerId}>
Original Price: $${originalPrice}
Discount Amount: $${amount}
New Price: $${discountedPrice}
Reason: ${reason}
      `;

      await discountChannel.send({ content: messageText });

      // Ping the customer
      await discountChannel.send({ content: `<@${customerId}> You got a discount!` });

      await interaction.reply({ content: "Discount applied successfully!", ephemeral: true });
    } catch (err) {
      console.error("Discount handler error:", err);
      await interaction.reply({ content: "Failed to apply discount.", ephemeral: true });
    }
  });
};
